src/create.rs

```diff
@@ -50,25 +62,25 @@ impl Party0 {
             X: X_other,
             tid: tid_other,
         }: Message0,
-    ) -> Party1 {
+    ) -> anyhow::Result<Party1> {
         let r = RevocationKeyPair::new_random();
         let y = PublishingKeyPair::new_random();
 
         let TX_f = FundingTransaction::new(
```

Comment:
Not related to this PR: 

I think we will need to split up Party0 in 2 parties A+B as we need both parties to have the same `TX_f` at the end. Right now TX_f either spends into an address `A+B` or `B+A`, i.e. it does make a difference which pk is `X_other` and `X_self` (check out the test).